clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -166,33 +167,13 @@ public void testMembershipManagerInitSupportsEmptyGroupInstanceId() {
         createMembershipManagerJoiningGroup();
     }
 
-    @Test
-    public void testMembershipManagerRegistersForClusterMetadataUpdatesOnFirstJoin() {
```

Comment:
this is just small clean-up unrelated to this PR (but I found it while on it). This test does not apply anymore, and it passes only because it's not really testing what its name suggest. 

The `membershipMgr` did register for metadata updates at some point, but it stopped doing it the moment it was turned into a manager that could be polled

Reply:
we're partners in crime here :) I totally missed this in reviews